cmd/sonobuoy/app/pluginCache.go

```diff
@@ -0,0 +1,266 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/vmware-tanzu/sonobuoy/pkg/plugin/loader"
+	"github.com/vmware-tanzu/sonobuoy/pkg/plugin/manifest"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	SonobuoyDirEnvKey = "SONOBUOY_DIR"
+)
+
+var (
+	defaultSonobuoyDir = filepath.Join("~", ".sonobuoy")
+)
+
+func NewCmdPlugin() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "plugin",
+		Aliases: []string{"plugins"},
+		Short:   "Manage your installed plugins",
+		Hidden:  true,
+	}
+
+	listCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all installed plugins",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return listInstalledPlugins(getPluginCacheLocation(cmd))
+		},
+	}
+
+	showCmd := &cobra.Command{
+		Use:   "show <plugin filename>",
+		Short: "Print the full definition of the named plugin file",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return showInstalledPlugin(getPluginCacheLocation(cmd), filenameFromArg(args[0], ".yaml"))
+		},
+	}
+
+	installCmd := &cobra.Command{
+		Use:   "install <save-as-filename> <source filename or URL>",
+		Short: "Install a plugin so that it can be run via just its filename rather than a full path or URL.",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return installPlugin(getPluginCacheLocation(cmd), filenameFromArg(args[0], ".yaml"), args[1])
+		},
+	}
+
+	uninstallCmd := &cobra.Command{
+		Use:   "uninstall <plugin filename>",
+		Short: "Uninstall a plugin. You can continue to run any plugin via specifying a file or URL.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deletePlugin(getPluginCacheLocation(cmd), filenameFromArg(args[0], ".yaml"))
+		},
+	}
+
+	cmd.AddCommand(listCmd, showCmd, installCmd, uninstallCmd)
+
+	return cmd
+}
+
+// getPluginCacheLocation will return the location of the plugin cache (defaults to ~/.sonobuoy)
+// If no override is set in the env, the default is assumed. If we are unable to expand to an abosolute
+// path then we return the empty string signalling the feature should be disabled.
+func getPluginCacheLocation(cmd *cobra.Command) string {
+	usePath := os.Getenv(SonobuoyDirEnvKey)
+	if len(usePath) == 0 {
+		usePath = defaultSonobuoyDir
+	}
+	expandedPath, err := homedir.Expand(usePath)
```

Comment:
Suggestion: you could save a dependency here. If I recall correctly, that `home-dir` package essentially does the following.

```go
func ExpandPath(path string) (string, error) {
	home, err := os.UserHomeDir()
	if err != nil {
		return "", err
	}

	if path[0] == '~' {
		path = filepath.Join(home, path[1:])
	}
	return path, nil
}
```

Reply:
Maybe I will; originally I used home-dir since it mentioned something about cgo but that was very outdated. https://github.com/mitchellh/go-homedir/issues/20#issuecomment-467476994 mentions the stdlib vs home-dir usage and I may just prefer the stdlib in our case.